pkg/proxy/proxy/transparent.go

```diff
@@ -0,0 +1,188 @@
+package proxy
+
+import (
+	"bufio"
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+	"net/url"
+	"regexp"
+	"sync"
+
+	"github.com/elazarl/goproxy"
+	"github.com/google/oss-rebuild/internal/proxy/handshake"
+	"github.com/google/oss-rebuild/pkg/proxy/cert"
+	"github.com/google/oss-rebuild/pkg/proxy/logger"
+)
+
+// TLS port to which proxied TLS traffic should be redirected.
+// Used to enable customization during testing.
+var tlsPort = "443"
+
+// TransparentProxyServer transparently proxies HTTP and HTTPS traffic.
+type TransparentProxyServer struct {
```

Comment:
One idea might be to change this name to service instead of server.  that way you wouldn't need to change the  name of the other constructor.